source/extensions/transport_sockets/s2a/config.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "envoy/server/transport_socket_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace S2A {
+
+// S2A config registry
+class S2ATransportSocketConfigFactory
+    : public virtual Server::Configuration::TransportSocketConfigFactory {
+public:
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override;
+  std::string name() const override { return "envoy.transport_sockets.s2a"; }
```

Comment:
Is there any reason why the implementation of the `name()` method is in the .h file rather than the .cc?

Reply:
No particular reason outside of mimicking the existing ALTS implementation.

Reply:
Could we move it to the .cc file to be consistent with the rest of the methods?